go/vt/vttablet/tabletserver/vstreamer/rowstreamer.go

```diff
@@ -237,11 +240,9 @@ func (rs *rowStreamer) streamQuery(conn *snapshotConn, send func(*binlogdatapb.V
 	byteCount := 0
 	for {
 		//log.Infof("StreamResponse for loop iteration starts")
-		select {
```

Comment:
Will we always get ctx.Err() as not nil when the context expires? We just call ctx.Done() on timeouts, for example.

Reply:
@rohit-nayak-ps: Yes, you can verify the implementation on https://golang.org/src/context/context.go#L394 -- this is a known performance fix. Doing a `default` read on the `Done()` channel is always equivalent to checking `Err()`, because the only implementation for this functionality is in the `cancelCtx` which behaves this way. It doesn't matter if this is a manually canceled context, timeout or duration context.

Reply:
Ah, yes of course, it sets the err to Cancelled